### PR TITLE
[BUGFIX] fix wrong type and pass correct configuration object

### PR DESCRIPTION
**Pull request by kaystrobach**

### CODE REVIEW

File: Classes/ViewHelpers/ImageViewHelper.php [hunk 1]
```diff
@@ -57,14 +60,12 @@ public function initializeArguments()
...
-        $thumbnailImage = $this->getThumbnailImage($identicon->getImage(), $size);
+        $identicon = $this->identiconFactory->create(new IdenticonConfiguration($hash, $size));
 
         $this->tag->addAttributes(array(
-            'width' => $thumbnailImage->getWidth(),
-            'height' => $thumbnailImage->getHeight(),
-            'src' => $this->resourcePublisher->getPersistentResourceWebUri($thumbnailImage->getResource()),
+            'width' => $size,
+            'height' => $size,
+            'src' => 'data:image/png;base64,' . base64_encode($identicon),
```
Comment (dfeyer) [inline Classes/ViewHelpers/ImageViewHelper.php:68]: The base64 encoding can be optional ? What do you think to introduce a new argument in the VH ? i'm fine it's enabled by default, but can be disabled if needed.

**Comment by kaystrobach** [comment]
the VH failed for me, so i debugged it and as the resoult returned was no resource, but an image as string i used the base64 function to get a valid output

File: Classes/ViewHelpers/ImageViewHelper.php (hunk 1 above)
Comment (kaystrobach) [inline Classes/ViewHelpers/ImageViewHelper.php:68]: the VH failed for me, so i debugged it and as the resoult returned was no resource, but an image as string.So  i used the base64 function to get a valid output.

I'm also find with returning an image path here.
As the generated images are usually very small i also think that the encoding is a the proper way to go.

**Comment by dfeyer** [comment]
Thanks, fine for this one, maybe we can introduce a setting in a future release

**Pull request merged**